buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -160,6 +160,8 @@ class BuildPlugin implements Plugin<Project> {
                             testClusters.all { ElasticsearchCluster cluster ->
                                 cluster.systemProperty 'javax.net.ssl.trustStorePassword', 'password'
                                 cluster.systemProperty 'javax.net.ssl.keyStorePassword', 'password'
+                                // Can't use our DiagnosticTrustManager with SunJSSE in FIPS mode
+                                cluster.setting 'xpack.security.ssl.diagnose.trust', 'false'
```

Comment:
Is this safe to do all the time, even when not running in FIPS mode?

Also, if this is required when running in FIPS mode, how is this set for users in production? Do they have to manually add this setting?

Reply:
Yes, it's always safe, it just removes some useful diagnostics (that we want on by default).

https://github.com/elastic/elasticsearch/pull/49888 will disable it automatically in Elasticsearch's FIPS mode on 7.x (we don't need to disable it on master, because FIPS works differently in JDK9+).

However, we don't currently enable Elasticsearch-FIPS mode for every test cluster that runs on a FIPS JVM, so we want to explicitly disable diagnostics.